test/contrib/test_ssltransport.py

```diff
@@ -0,0 +1,402 @@
+from dummyserver.testcase import SocketDummyServerTestCase, consume_socket
+from dummyserver.server import (
+    DEFAULT_CERTS,
+    DEFAULT_CA,
+)
+
+from urllib3.contrib.ssl import SSLTransport
+
+import select
+import pytest
+import socket
+import ssl
+import sys
+
+
+def server_client_ssl_contexts():
+    if hasattr(ssl, "PROTOCOL_TLS_SERVER"):
+        server_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+    else:
+        server_context = ssl.SSLContext(ssl.PROTOCOL_TLS)
+    server_context.load_cert_chain(DEFAULT_CERTS["certfile"], DEFAULT_CERTS["keyfile"])
+
+    if hasattr(ssl, "PROTOCOL_TLS_CLIENT"):
+        client_context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+    else:
+        client_context = ssl.SSLContext(ssl.PROTOCOL_TLS)
+        client_context.verify_mode = ssl.CERT_REQUIRED
+        client_context.check_hostname = True
+
+    client_context.load_verify_locations(DEFAULT_CA)
+    return server_context, client_context
+
+
+def sample_request(binary=True):
+    request = (
+        b"GET http://www.testing.com/ HTTP/1.1\r\n"
+        b"Host: www.testing.com\r\n"
+        b"User-Agent: awesome-test\r\n"
+        b"\r\n"
+    )
+    return request if binary else request.decode("utf-8")
+
+
+def validate_request(provided_request, binary=True):
+    assert provided_request is not None
+    expected_request = sample_request(binary)
+    assert provided_request == expected_request
+
+
+def sample_response(binary=True):
+    response = b"HTTP/1.1 200 OK\r\n" b"Content-Length: 0\r\n" b"\r\n"
```

Comment:
This is three separate `bytes` instances joining together, could we change this to `b"HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n"